test/conftest.py

```diff
@@ -478,70 +485,124 @@ def get_retriever(retriever_type, document_store):
 
 
 @pytest.fixture(params=["elasticsearch", "faiss", "memory", "milvus", "weaviate"])
-def document_store_with_docs(request, test_docs_xs):
-    document_store = get_document_store(request.param)
-    document_store.write_documents(test_docs_xs)
-    yield document_store
-    document_store.delete_documents()
-
+def document_store_with_docs(request, test_docs_xs, tmp_path):
+    for document_store in get_document_store(request.param, tmp_path=tmp_path):
```

Comment:
Yep this is the ugliness I was talking about. In order to be able to use `get_document_store` as a function in both tests and other fixtures, I needed to convert the `return` into a `yield`. That made it effectively a generator, and using the `for` is an (extremely ugly, but) effective way to make sure that the teardown part included in `get_document_store` will be executed at the right time (that is, when leaving the `for` block).

I'd never think this is good code, but I didn't want to spend more time for a proper, way more complex solution, given how confusing the document store fixtures are right now. I think it would be wasted time tbh. I'd rather introduce it here and then remove it ASAP along with a more comprehensive refactoring of the docstore tests.

Reply:
Ok makes sense. Ideally, we find a way to get rid of the direct usage of `get_document_store` and just use a standardized way with fixtures :thinking: 

Can you then please add a comment in the meantime and maybe already create an issue for "refactoring document store fixtures" that we can link here?